client/src/app/site/common/components/user-statistics/user-statistics.component.ts

```diff
@@ -0,0 +1,221 @@
+
+    /**
+     * Holds information about hours, minutes and seconds for the total duration of word requests.
+     */
+    private durationWordRequest: TimeObject = null;
+
+    /**
+     * List of unique speakers.
+     */
+    private uniqueSpeakers: ViewSpeaker[] = [];
+    private speakersOfStructureLevel: Map<string, SpeakersOfStructureLevel> = new Map();
+    private _numberOfWordRequests = 0;
+    private statisticIsOpen = false;
+    private speakersOfStructureLevelSubject = new BehaviorSubject<SpeakersOfStructureLevel[]>([]);
+
+    public constructor(
+        title: Title,
+        protected translate: TranslateService,
+        matSnackBar: MatSnackBar,
+        private losRepo: ListOfSpeakersRepositoryService,
+        private cd: ChangeDetectorRef
+    ) {
+        super(title, translate, matSnackBar);
+    }
+
+    public ngOnInit(): void {
+        this.subscriptions.push(
+            this.losRepo.getViewModelListObservable().subscribe(losList => {
+                this.calcWordRequests(losList.filter(los => los.closed));
+                this.pushNextState();
+            })
+        );
+    }
+
+    /**
+     * Opens or closes statistics.
+     */
+    public changeViewOfStatistics(): void {
+        this.statisticIsOpen = !this.statisticIsOpen;
+    }
+
+    /**
+     * This iterates over a list of list-of-speakers. For each speaker it calculates the duration the speaker
+     * has spoken.
+     *
+     * @param listOfSpeakers List of list-of-speakers.
+     */
+    private calcWordRequests(listOfSpeakers: ViewListOfSpeakers[]): void {
+        let durationWordRequest = 0;
+        let numberOfWordRequests = 0;
+        for (const los of listOfSpeakers) {
+            los.speakers.forEach(speaker => {
+                const duration = this.getNumberOfDate(speaker);
+                durationWordRequest += duration;
+                this.addUniqueSpeaker(speaker);
+                this.addSpeakerToStructureLevel(speaker, duration);
+            });
+            numberOfWordRequests += los.speakers.length;
+        }
+        this.durationWordRequest = this.getTime(durationWordRequest);
+        this._numberOfWordRequests = numberOfWordRequests;
+    }
+
+    /**
+     * Iterates over map of speakers-information and pushes a next state to the subject of speakers-information.
+     */
+    private pushNextState(): void {
+        for (const structureLevel of this.speakersOfStructureLevel.keys()) {
+            const structureLevelObject = this.speakersOfStructureLevel.get(structureLevel);
+            structureLevelObject.durationString = this.parseDuration(this.getTime(structureLevelObject.durationNumber));
+            this.speakersOfStructureLevel.set(structureLevel, structureLevelObject);
+        }
+        this.speakersOfStructureLevelSubject.next(Array.from(this.speakersOfStructureLevel.values()));
+        this.cd.detectChanges();
+    }
+
+    private addUniqueSpeaker(speaker: ViewSpeaker): boolean {
```

Comment:
the `list-of-speaker-repository` can basically get this number for you.
see `isFirstContribution` as a reference. Basically, get all finishedSpeakers in all lists, map them to their unique user Id's and create an `set` using `[...new Set(myArray)];`

Try to use existing services and avoid putting too much logic into components.

Reply:
I solved it in a slightly different way. A speaker is pushed immediately to a resulting array, if not already existing.